File: hrms/src/main/java/com/nberimen/candidate/CandidateService.java
package com.nberimen.candidate;

import java.util.List;

import com.nberimen.resume.Resume;
import com.nberimen.utilities.result.DataResult;
import com.nberimen.utilities.result.Result;

public interface CandidateService {

	public DataResult<List<Candidate>> getAll();
	public Result register(Candidate candidate);
	public Result saveResume(Resume resume, int candidateId);
	public DataResult<Candidate> getById(int id);
}

File: hrms-frontend/src/services/candidateService.js
import axios from 'axios'
export default class CandidateService {
    registerCandidate = candidate => {
        return axios.post("http://localhost:8080/api/candidates/register", candidate);
    }
    getAllCandidates = () => {
        return axios.get("http://localhost:8080/api/candidates/getall");
    }
    createResume = (resume, candidateId) => {
        return axios.post(`http://localhost:8080/api/candidates/createresume/${candidateId}`, resume);

    }
    getById=(id)=>{
        return axios.get("http://localhost:8080/api/candidates/getById?id="+id)
    }
}
